plutus-core/changelog.d/20230724_195443_effectfully_make_elaboration_of_builtins_customizable.md

```diff
@@ -0,0 +1,3 @@
+### Changed
+
+- Separated the single `Includes` constraint into two constraints, `HasTypeLevel` and `HasTermLevel` (which together form `HasBothLevel`) in #5434.
```

Comment:
> Will we have something like `DefaultTypeUni` and `DefaultTermUni`

Yup. We might want to choose different names though. I'd really like to replace all `Default` business with `Cardano` for example.

> `(DefaultTypeUni HasType a, DefaultTermUni HasTerm a, ...)`

We still want to reference a single `uni` variable, because adding another type variable isn't going to appeal to anyone. We're probably going to associate a data family with that `uni` containing the type-level counterparts of types whose values are available at the term level (one of the worst sentences I've ever written, sorry about that). This way people who only care about the term level (anyone targeting UPLC) will not need to know anything about the type-level universe.

You can see a [prototype](https://github.com/input-output-hk/plutus/compare/effectfully/builtins/treating-type-heads?expand=1#diff-4c05db6daf04fdffe5653ce0426d2707197302397c7382e97f1b1c745973be2a) of the idea, but it's a bit old.